src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java

```diff
@@ -109,16 +126,52 @@ private void getConnection(MailMessage message, Handler<AsyncResult<MailResult>>
     });
   }
 
+  private Future<Void> dkimFuture(Context context, EncodedPart encodedPart) {
+    Promise<Void> promise = Promise.promise();
+    List<Future> dkimFutures = new ArrayList<>();
+    // run dkim sign, and add email header after that.
+    dkimSigners.forEach(dkim -> dkimFutures.add(dkim.signEmail(context, encodedPart)));
+    CompositeFuture.all(dkimFutures).setHandler(r -> {
```

Comment:
instead you can do:

```
Composite.all(list).map(f)
```
 instead of creating a Promise and completing it in the CompositeFuture handler

Reply:
:+1:  Done by using `map(f)` instead of creating a new Promise.